resources/sts/4.12/hypershift/InstallerPolicy.json

```diff
@@ -4,70 +4,146 @@
+                "ec2:CreateTags",
+                "ec2:DeleteTags",
                 "ec2:DescribeAvailabilityZones",
                 "ec2:DescribeInternetGateways",
-                "ec2:DescribeNetworkInterfaces",
                 "ec2:DescribeRegions",
                 "ec2:DescribeReservedInstancesOfferings",
                 "ec2:DescribeRouteTables",
-                "ec2:DescribeSecurityGroups",
                 "ec2:DescribeSecurityGroupRules",
                 "ec2:DescribeSubnets",
                 "ec2:DescribeVpcAttribute",
                 "ec2:DescribeVpcs",
-                "ec2:RunInstances",
-                "elasticloadbalancing:CreateLoadBalancer",
+                "ec2:DescribeInstanceTypeOfferings",
                 "elasticloadbalancing:DeleteLoadBalancer",
-                "iam:AddRoleToInstanceProfile",
-                "iam:AttachRolePolicy",
-                "iam:CreateInstanceProfile",
-                "iam:CreateOpenIDConnectProvider",
-                "iam:CreatePolicy",
-                "iam:CreatePolicyVersion",
-                "iam:CreateRole",
-                "iam:DeleteOpenIDConnectProvider",
-                "iam:DeletePolicyVersion",
-                "iam:DeleteRole",
-                "iam:DeleteRolePermissionsBoundary",
-                "iam:DetachRolePolicy",
-                "iam:GetOpenIDConnectProvider",
-                "iam:GetPolicy",
-                "iam:GetRole",
-                "iam:GetUser",
-                "iam:ListAttachedRolePolicies",
-                "iam:ListPolicyTags",
-                "iam:ListPolicyVersions",
-                "iam:ListRoleTags",
-                "iam:PutRolePermissionsBoundary",
-                "iam:TagInstanceProfile",
-                "iam:TagPolicy",
                 "kms:DescribeKey",
-                "route53:ChangeResourceRecordSets",
-                "route53:DeleteHostedZone",
-                "route53:GetAccountLimit",
-                "route53:GetHostedZone",
+                "iam:GetRole",
                 "route53:ListHostedZones",
-                "route53:ListHostedZonesByName",
                 "route53:ListResourceRecordSets",
-                "s3:CreateBucket",
-                "s3:DeleteBucket",
-                "s3:DeleteObject",
-                "s3:ListObjects",
-                "s3:PutObject",
-                "secretsmanager:GetSecretValue",
+                "route53:GetAccountLimit",
                 "servicequotas:GetServiceQuota",
-                "sts:AssumeRole",
-                "sts:AssumeRoleWithWebIdentity",
-                "sts:GetCallerIdentity"
+                "sts:GetCallerIdentity",
+                "iam:TagInstanceProfile"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Action": [
+                "iam:PassRole"
+            ],
+            "Resource": [
+                "arn:*:iam::*:role/*-Worker-Role"
+            ],
+            "Effect": "Allow",
+            "Condition": {
```

Comment:
It is possible to also add a condition here that the role has the red-hat-managed: true tag? I'm not sure if you need a new SID for that

Reply:
The `PassRolw` doesn't support tag based condition keys according to the [docs](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsidentityandaccessmanagement.html)